core/api/views/business_plan.py

```diff
@@ -79,6 +79,33 @@ def get_years(self, *args, **kwargs):
             )
         )
 
+    @action(methods=["GET"], detail=False)
+    def get(self, *args, **kwargs):
+        bp = get_business_plan_from_request(self.request)
+        history_qs = bp.bphistory.select_related("business_plan", "updated_by")
+        activities = bp.records.select_related(
+            "business_plan",
+            "business_plan__agency",
+            "country",
+            "sector",
+            "subsector",
+            "project_type",
+            "bp_chemical_type",
+            "project_cluster",
+        ).prefetch_related(
+            "substances",
+            "blends",
+            "values",
+        )
+
+        return Response(
+            {
+                "business_plan": BusinessPlanSerializer(bp).data,
+                "history": BPHistorySerializer(history_qs, many=True).data,
+                "activities": BPRecordDetailSerializer(activities, many=True).data,
```

Comment:
The problem with this serializer is that it will add business plan details for each activity, and here we do not need it. 
You can add a context variable and check it before you add the bp details or you can create another serializer